src/strategies/convex/strategy-convex-stecrv.sol

```diff
@@ -0,0 +1,141 @@
+
+contract StrategyConvexSteCRV is StrategyConvexFarmBase  {
+    address public stEth = 0xae7ab96520DE3A18E5e111B5EaAb095312D7fE84;
+    address public lpToken = 0x06325440D014e39736583c165C2963BA99fAf14E; // steCRV
+
+    address public pool = 0xDC24316b9AE028F1497c275EB9192a3Ea0f67022; // Curve ETH-stETH
+
+    // extra reward
+    address public constant ldo = 0x5A98FcBEA516Cf06857215779Fd812CA3beF1B32;
+
+    uint256 public constant cvxPoolId = 25;
+
+    // How much CVX tokens to keep
+    uint256 public keepCVX = 500;
+    uint256 public keepCVXMax = 10000;
+
+    constructor(
+        address _governance,
+        address _strategist,
+        address _controller,
+        address _timelock
+    )
+        public
+        StrategyConvexFarmBase(
+            lpToken,
+            cvxPoolId,
+            _governance,
+            _strategist,
+            _controller,
+            _timelock
+        )
+    {
+    }
+
+    // swap for eth
+    receive() external payable {}
+
+    // **** Getters ****
+
+    function getName() external pure override returns (string memory) {
+        return "StrategyConvexStETH";
+    }
+
+    function getHarvestable() public view returns (uint256, uint256, uint256, uint256) {
+        return (get_crv_earned(), get_cvx_earned(), get_ldo_earned(), get_eth_earned());
+    }
+
+    function get_ldo_earned() public view returns (uint256) {
+        return IVirtualBalanceRewardPool(IBaseRewardPool(getCrvRewardContract()).extraRewards(0)).earned(address(this));
+    }
+
+    function get_eth_earned() public view returns (uint256) {
+        uint256 claimableCrv = get_crv_earned();
+        uint256 claimableLdo = get_ldo_earned();
+        uint256 claimableCvx = get_cvx_earned();
+
+        uint256 crvEth = _estimateSell(crv, claimableCrv);
+        uint256 ldoEth = _estimateSell(ldo, claimableLdo);
+        uint256 cvxEth = _estimateSell(cvx, claimableCvx);
+
+        return crvEth.add(ldoEth).add(cvxEth);
+    }
+
+    function _estimateSell(address currency, uint256 amount) internal view returns (uint256 outAmount){
+        address[] memory path = new address[](2);
+        path[0] = currency;
+        path[1] = weth;
+        uint256[] memory amounts = UniswapRouterV2(sushiRouter).getAmountsOut(amount, path);
+        outAmount = amounts[amounts.length - 1];
+
+        return outAmount;
+    }
+
+    // **** Setters ****
+
+    function setKeepCVX(uint256 _keepCVX) external {
+        require(msg.sender == governance, "!governance");
+        keepCVX = _keepCVX;
+    }
+
+    // **** State Mutations ****
+
+    function harvest() public override onlyBenevolent {
+        IBaseRewardPool(getCrvRewardContract()).getReward(
+            address(this),
+            true
+        );
+
+        uint256 _cvx = IERC20(cvx).balanceOf(address(this));
+        if (_cvx > 0) {
+            // How much CVX to keep to restake?
+            uint256 _keepCVX = _cvx.mul(keepCVX).div(keepCVXMax);
```

Comment:
@shortstackdev can you help me understand the reasoning behind this calculation?

Reply:
This was a port from strategies/curve/strategy-curve-stecrv.sol, replacing keepCRV with keepCVX.

The currently deployed strat https://etherscan.io/address/0x350c4f7a669dc263ec1838fa105172e1d96e8259#readContract has keepCRV = 500 and keepCRVMax = 10000, so the stEth jar is withholding 5% of CRV rewards - and this is on top of performance fees.

The intent was to withhold a portion of the CRV rewards to lock for veCRV to maintain boost. I guess that no longer applies, but I left this in, in case the protocol wants to hold on to CVX to vote for gauges within Convex.